# Libraries

In [0]:
!pip install plotly==4.8.1
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
pd.options.plotting.backend = "plotly"

# Data Base

In [0]:
df = pd.read_csv('/content/drive/My Drive/ML Projects/Womens Clothing E-Commerce/Womens Clothing E-Commerce Reviews.csv')
df

Unnamed: 0  Clothing ID  Age  ...   Division Name Department Name  Class Name
0               0          767   33  ...       Initmates        Intimate   Intimates
1               1         1080   34  ...         General         Dresses     Dresses
2               2         1077   60  ...         General         Dresses     Dresses
3               3         1049   50  ...  General Petite         Bottoms       Pants
4               4          847   47  ...         General            Tops     Blouses
...           ...          ...  ...  ...             ...             ...         ...
23481       23481         1104   34  ...  General Petite         Dresses     Dresses
23482       23482          862   48  ...  General Petite            Tops       Knits
23483       23483         1104   31  ...  General Petite         Dresses     Dresses
23484       23484         1084   28  ...         General         Dresses     Dresses
23485       23485         1104   52  ...  General Petite         Dresses     Dresses

[23486 rows x 11 columns]

## Content
This dataset includes 23486 rows and 10 feature variables. Each row corresponds to a customer review, and includes the variables:

- Clothing ID: Integer Categorical variable that refers to the specific piece being reviewed.
- Age: Positive Integer variable of the reviewers age.
- Title: String variable for the title of the review.
- Review Text: String variable for the review body.
- Rating: Positive Ordinal Integer variable for the product score granted by the customer from 1 Worst, to 5 Best.
- Recommended IND: Binary variable stating where the customer recommends the product where 1 is recommended, 0 is not recommended.
- Positive Feedback Count: Positive Integer documenting the number of other customers who found this review positive.
- Division Name: Categorical name of the product high level division.
- Department Name: Categorical name of the product department name.
- Class Name: Categorical name of the product class name.



# Data Vizualization 

## Clothing ID

In [0]:
df['Clothing ID'].plot.hist(title='QTY Sales by Clothing ID',nbins=1300)

## Age

In [0]:
df['Age'].plot.hist(title = 'Age Distibution')

## Rating

In [0]:
df['Rating'].plot.hist(title = 'Rating Distribution')

In [0]:
aux = df.groupby(['Recommended IND','Rating']).size().reset_index().rename(columns={0:'count'})
aux['Rating'] = aux['Rating'].astype(str)
fig = px.bar(aux, x='Recommended IND', y='count', color='Rating')
fig.update_layout(title = 'Recommended IND by Rating')
fig.show()

## Recommended IND

In [0]:
df['Recommended IND'].plot.hist(title = 'Recommended IND Distribution')

## Positive Feedback Count

In [0]:
aux = df.groupby(['Rating']).sum().reset_index()
aux['Rating'] = aux['Rating'].astype(str)
fig = px.bar(aux, x='Rating', y='Positive Feedback Count')
fig.update_layout(title = 'QTY. Of Positive Feedback by Rating')
fig.show()

## Division Name

In [0]:
df['Division Name'].plot.hist(title = 'Division Name Distribution')

In [0]:
fig = px.bar(df.groupby(['Rating','Division Name']).size().reset_index().rename(columns={0:'count'}), x="Rating", y="count", color='Division Name')
fig.update_layout(title = 'Rating by Division Name')
fig.show()

In [0]:
fig = px.bar(df.groupby(['Recommended IND','Division Name']).size().reset_index().rename(columns={0:'count'}), x="Recommended IND", y="count", color='Division Name')
fig.update_layout(title = 'Recommended IND by Division Name')
fig.show()

## Department Name

In [0]:
df['Department Name'].plot.hist(title = 'Departament Name Distibution')

In [0]:
fig = px.bar(df.groupby(['Division Name','Department Name']).size().reset_index().rename(columns={0:'count'}), x="Division Name", y="count", color='Department Name')
fig.update_layout(title = 'Division Name by Department Name')
fig.show()

In [0]:
fig = px.bar(df.groupby(['Rating','Department Name']).size().reset_index().rename(columns={0:'count'}), x="Rating", y="count", color='Department Name')
fig.update_layout(title = 'Rating by Department Name')
fig.show()

In [0]:
fig = px.bar(df.groupby(['Recommended IND','Department Name']).size().reset_index().rename(columns={0:'count'}), x="Recommended IND", y="count", color='Department Name')
fig.update_layout(title = 'Recommended IND by Department Name')
fig.show()

## Class Name

In [0]:
df['Class Name'].hist(title = 'Class Distribution')

In [0]:
fig = px.bar(df.groupby(['Division Name','Class Name']).size().reset_index().rename(columns={0:'count'}), x="Division Name", y="count", color='Class Name')
fig.update_layout(title = 'Division Name by Class Name')
fig.show()

In [0]:
fig = px.bar(df.groupby(['Department Name','Class Name']).size().reset_index().rename(columns={0:'count'}), x="Department Name", y="count", color='Class Name')
fig.update_layout(title = 'Department Name by Class Name')
fig.show()

In [0]:
fig = px.bar(df.groupby(['Rating','Class Name']).size().reset_index().rename(columns={0:'count'}), x="Rating", y="count", color='Class Name')
fig.update_layout(title = 'Rating by Class Name')
fig.show()

In [0]:
fig = px.bar(df.groupby(['Recommended IND','Class Name']).size().reset_index().rename(columns={0:'count'}), x="Recommended IND", y="count", color='Class Name')
fig.update_layout(title = 'Recommended IND by Class Name')
fig.show()